galsim/wfirst/wfirst_wcs.py

```diff
@@ -0,0 +1,571 @@
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+"""
+@file wfirst_wcs.py
+
+Part of the WFIRST module.  This file includes any routines needed to define and use the WFIRST WCS.
+"""
+import galsim
+import galsim.wfirst
+import numpy as np
+import os
+
+# Basic WFIRST reference info, with lengths in mm.
+pixel_size_mm = 0.01
+focal_length = 18500.
+pix_scale = (pixel_size_mm/focal_length)*galsim.radians
+n_sip = 5 # Number of SIP coefficients used, where arrays are n_sip x n_sip in dimension
+
+# Version-related information, for reference back to material provided by Jeff Kruk.
+tel_name = "WFIRST"
+instr_name = "WFI"
+optics_design_ver = "4.2.2"
+prog_version = "0.4"
+
+# Information about center points of the SCAs in the WFI focal plane coordinate system (f1, f2)
+# coordinates.  These are rotated by an angle theta_fpa with respect to the payload axes, as
+# projected onto the sky. Note that the origin is at the center of the payload axes, on the
+# telescope boresight, not centered on the FPA.  Since the SCAs are 1-indexed, these arrays have a
+# non-used entry with index 0.  i.e., the maximum SCA is 18, and its value is in sca_xc_mm[18].
+# Units are mm.
+sca_xc_mm = np.array([0., -21.690, -21.690, -21.690, -65.070, -65.070, -65.070, -108.450, -108.450,
+                      -108.450,  21.690,  21.690,  21.690,  65.070,  65.070,  65.070, 108.450,
+                      108.450, 108.450])
+sca_yc_mm = np.array([0., 199.250, 149.824, 242.630, 188.250, 138.824, 231.630, 164.923, 115.498,
+                      208.304, 199.250, 149.824, 242.630, 188.250, 138.824, 231.630, 164.923,
+                      115.498, 208.304])
+# Nominal center of FPA in this coordinate frame, in mm and as an angle.
+fpa_xc_mm = 0.0
+fpa_yc_mm = 199.250
+xc_fpa = np.arctan(fpa_xc_mm/focal_length)*galsim.radians
+yc_fpa = np.arctan(fpa_yc_mm/focal_length)*galsim.radians
+
+# The next array contains rotation offsets of individual SCA Y axis relative to FPA f2 axis. Same
+# sign convention as theta_fpa. These represent mechanical installation deviations from perfect
+# alignment and are ideally zero. These will be measured during focal plane integration and
+# testing.
+sca_rot = np.zeros_like(sca_xc_mm)
+
+# Rotation of focal plane axes relative to payload axes: positive CCW rotation of the f2 axis
+# relative to -Y_pl, and of f1 relative to +X_pl.
+theta_fpa = 32.5*galsim.degrees
+
+# File with SIP coefficients.
+sip_filename = os.path.join(galsim.meta_data.share_dir, 'sip_422.txt')
+
+def getWCS(PA, ra=None, dec=None, pos=None, PA_is_FPA=False, as_header=False):
+    """
+    This routine gets a list of WCS, one for each of the WFIRST SCAs (Sensor Chip Array, the
+    equivalent of a chip in an optical CCD).  Since the WFIRST SCAs are labeled 1-18, the zeroth
+    list item is simply None.  Use of this routine requires that GalSim be able to access some
```

Comment:
I'm not so fond of this returning a list.  As with `getPSF` I recommend returning a dict that is indexed by SCA.  Also, to get parallel with that, I recommend letting it take in an `SCAs` parameter to say which keys you want in the returned dict.